formatter/src/main/java/org/togetherjava/formatter/tokenizer/TokenType.java

```diff
@@ -0,0 +1,127 @@
+package org.togetherjava.formatter.tokenizer;
+
+import java.util.regex.Pattern;
+
+/**
+ * Represents every possible token that can be lexed by the lexer
+ *
+ * @author illuminator3
+ */
+public enum TokenType {
+    // keywords
+    CLASS("class", true),
+    ENUM("enum", true),
+    RECORD("record", true),
+    INTERFACE("interface", true),
+    IMPORT("import", true),
+    IF("if", true),
+    FOR("for", true),
+    WHILE("while", true),
+    NULL("null", true),
+    EXTENDS("extends", true),
+    IMPLEMENTS("implements", true),
+    NEW("new", true),
+    RETURN("return", true),
+    PACKAGE("package", true),
+    THIS("this", true),
+    YIELD("yield", true),
+    CATCH("catch", true),
+    TRY("try", true),
+    ELSE_IF("else if", true), // not really a keyword but required for formatting
+    ELSE("else", true),
+    // access modifiers
+    PUBLIC("public", true),
+    PRIVATE("private", true),
+    PROTECTED("protected", true),
+    STATIC("static", true),
+    SEALED("sealed", true),
+    NON_SEALED("non-sealed", true),
+    FINAL("final", true),
+    ABSTRACT("abstract", true),
+    // primitive types
+    VOID("void", true),
+    INT("int", true),
+    LONG("long", true),
+    SHORT("short", true),
+    BYTE("byte", true),
+    BOOLEAN("boolean", true),
+    FLOAT("float", true),
+    DOUBLE("double", true),
+    CHAR("char", true),
+    // general tokens
+    OPEN_PARENTHESIS("("),
+    CLOSE_PARENTHESIS(")"),
+    OPEN_BRACES("{"),
+    CLOSE_BRACES("}"),
+    OPEN_BRACKETS("["),
+    CLOSE_BRACKETS("]"),
+    SEMICOLON(";"),
+    METHOD_REFERENCE("::"),
+    COLON(":"),
+    COMMA(","),
+    PLUS_EQUALS("+="),
+    MINUS_EQUALS("-="),
+    MULTIPLY_EQUALS("*="),
+    DIVIDE_EQUALS("/="),
+    MODULO_EQUALS("%="),
+    AND_EQUALS("&="),
+    OR_EQUALS("|="),
+    EQUALS("=="),
+    NOT_EQUALS("!="),
+    BIGGER_OR_EQUALS(">="),
+    BIGGER(">"),
+    SMALLER_OR_EQUALS("<="),
+    SMALLER("<"),
+    NOT("!"),
+    ASSIGN("="),
+    PLUSPLUS("++"),
+    PLUS("+"),
+    MINUSMINUS("--"),
+    MINUS("-"),
+    BOOL_AND("&&"),
+    BOOL_OR("||"),
+
+    // this has to be other DIVIDE or else it'll be seen as two divides
+    COMMENT(Pattern.compile("^(//.*|/\\*[^*]*\\*+(?:[^/*][^*]*\\*+)*/)")),
+
+    DIVIDE("/"),
+    MULTIPLY("*"),
+    DOT("."),
+    // other
+    ANNOTATION(Pattern.compile("^(@[a-zA-Z][a-zA-Z0-9_]*)")),
+    NUMBER(Pattern.compile("^((0[xb])?([0-9_]+|[0-9_]*(\\.\\d*))[dDfFlL]?)")),
+    STRING(Pattern.compile("^(\"[^\"]*\")")),
+    IDENTIFIER(Pattern.compile("^([a-zA-Z][a-zA-Z0-9_]*)")),
+    WHITSPACE(Pattern.compile("^(\s+)")),
+    UNKNOWN(Pattern.compile("^(.)"));
+
+    private final Pattern regex;
+    private final boolean isKeyword;
+
+    TokenType(String spattern, boolean isKeyword) {
+        if (!isKeyword) {
+            this.regex = Pattern.compile("^(\\Q" + spattern + "\\E)");
```

Comment:
Prefer properly named methods: `this.regex = Pattern.compile(Pattern.quote(spattern))` is easier to understand.